### ISSUE TITLE
Fix S4790 FN: Support multi-line string interpolation

### ISSUE THREAD

**Issue by gregory-paidis-sonarsource** [SonarSource/sonar-dotnet#6299]
### Description

FN for S4790 (CreatingHashAlgorithms).

Using C#11 multi-line string interpolation, the rule is not raising an issue.

### Repro steps

```csharp
string part1 = """System.Security.Cryptography""";
string part2 = """SHA1""";

void NewlinesInStringInterpolation()
{
    using var SHA1HashAlgorithm = HashAlgorithm.Create($"{part1 + // FN
        '.' +
        part2}");
    using var SHA1HashAlgorithmRawString = HashAlgorithm.Create($$"""{{part1 + // FN
        '.' +
        part2}}""");
}
```

### Expected behavior

The rule should raise an issue.

### Actual behavior

The rule does not raise an issue.
